src/commands/openYAMLConfigFile.ts

```diff
@@ -45,14 +47,67 @@ export async function openYAMLConfigFile(context: IActionContext, node?: StaticW
     await window.showTextDocument(configDocument, { selection });
 }
 
-async function getSelection(configDocument: TextDocument, buildConfigToSelect: BuildConfig): Promise<Range | undefined> {
-    const configRegex: RegExp = new RegExp(`${buildConfigToSelect}:`);
+/* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, no-prototype-builtins, @typescript-eslint/no-unsafe-assignment */
+export async function getSelection(configDocument: TextDocument, buildConfigToSelect: BuildConfig): Promise<Range | undefined> {
+    const configDocumentText: string = configDocument.getText();
+    const buildConfigRegex: RegExp = new RegExp(`${buildConfigToSelect}:`, 'g');
+    const buildConfigMatches: RegExpMatchArray | null = configDocumentText.match(buildConfigRegex);
 
-    let offset: number = configDocument.getText().search(configRegex);
-    // Shift the offset to the beginning of the build config's value
-    offset += `${buildConfigToSelect}: `.length;
+    if (buildConfigMatches && buildConfigMatches.length > 1) {
+        void ext.ui.showWarningMessage(localize('foundMultipleBuildConfigs', 'Multiple "{0}" build configurations were found in "{1}".', buildConfigToSelect, basename(configDocument.uri.fsPath)));
+        return undefined;
+    }
+
+    const parsedYaml: Document.Parsed = parseDocument(configDocumentText);
+    type YamlNode = YAMLMap | YAMLSeq | Pair | undefined;
+    const yamlNodes: YamlNode[] = [];
+    let yamlNode: YamlNode = parsedYaml.get('jobs');
```

Comment:
No need to hard-code "jobs" if we just use the root "contents"
```suggestion
    type YamlNode = YAMLMap | YAMLSeq | Pair | Scalar | undefined | null;
    const yamlNodes: YamlNode[] = [];
    let yamlNode: YamlNode = parsedYaml.contents;
```